web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -50,54 +47,46 @@ export class LayerDialogComponent implements OnDestroy {
   layer?: Layer;
   layerName!: string;
   projectId?: string;
-  activeProject$: Observable<Project>;
   subscription: Subscription = new Subscription();
   fieldTypes = FieldType;
   fields: List<Field>;
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) data: { layerId: string },
+    @Inject(MAT_DIALOG_DATA)
+    data: {
+      layerId: string;
+      projectId: string;
+      layers: Map<string, Layer>;
+      createLayer: boolean;
+    },
     private dialogRef: MatDialogRef<LayerDialogComponent>,
-    private projectService: ProjectService,
     private dataStoreService: DataStoreService,
     private router: Router,
     private confirmationDialog: MatDialog
   ) {
     this.lang = 'en';
     // Disable closing on clicks outside of dialog.
     dialogRef.disableClose = true;
+    this.fields = List<Field>();
```

Comment:
It looks like initialization of fields overlaps with `init()`. Inlining the init method here will make that clearer (see comment below).